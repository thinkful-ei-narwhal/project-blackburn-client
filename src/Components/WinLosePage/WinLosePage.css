.header {
  font-size: larger;
  margin: 20px;
  padding: 5px;
  background-color: #d9c1aa;
  box-shadow: 3px 3px 3px lightgray;
  width: 40%;
  text-align: center;
  margin-left: auto;
  margin-right: auto;
}

.game-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.btn-results {
  margin-top: 10px;
  margin-right: 10px;
  text-decoration: none;
  border: 1px solid #d9c1aa;
  padding: 20px;
  color: #302b28;
  border-radius: 6px;
  cursor: pointer;
}

.btn-results:hover {
  background-color: #ae9172;
  color: #d9c1aa;
}
