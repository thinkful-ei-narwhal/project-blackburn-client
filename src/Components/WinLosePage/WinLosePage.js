import React, { Component } from "react";
import { Link } from "react-router-dom";
import Button from "../Button/Button";
import ScoreboardApiService from "../../Services/scoreboard-api-service";
import "./WinLosePage.css";
import BlackBurnContext from "../../Context/BlackburnContext";

class WinLosePage extends Component {
  static contextType = BlackBurnContext;
  constructor(props) {
    super(props);

    this.state = {
      isShow: true,
      victory: this.props.win,
    };
  }

  postScore() {
    console.log(this.context.score)
    const data = {
      user_id: this.context.user.id,
      story_data: this.context.story_id,
      total_score: this.context.score,
      avg_wpm: 0,
      total_accuracy: 0,
    };
    console.log("postScore() with data", data);
    ScoreboardApiService.postScore(data);
  }

  handleRetryClick = () => {
    this.postScore();
    this.setState({ isShow: false });
  };

  handleNextClick = () => {
    if (this.context.incrementCheckpointIds() === null) {
      this.postScore();
      //redirect to FINAL win screen
    }
    this.setState({ isShow: false });
  };

  handleDashboardClick = () => {
    this.postScore();
    this.setState({ isShow: false });
  };

  renderWin() {
    return (
      <div className="results victory">
        <div className="results header">You're a genius bruh</div>
        <Link to={"/story"}>
          <Button
            className="btn results next-btn"
            onClick={this.handleNextClick}
          >
            Next
          </Button>
        </Link>
        <Link to="/dashboard">
          <Button
            className="btn results dashboard-btn"
            onClick={this.handleDashboardClick}
          >
            Dashboard
          </Button>
        </Link>
      </div>
    );
  }

  renderLose() {
    return (
      <div className="results defeat">
        <div className="results header">You suck and your guy died</div>
        <Link to={"/start"}>
          <Button
            className="btn results retry-btn"
            onClick={this.handleRetryClick}
          >
            Retry
          </Button>
        </Link>
        <Link to="/dashboard">
          <Button
            className="btn results dashboard-btn"
            onClick={this.handleDashboardClick}
          >
            Dashboard
          </Button>
        </Link>
      </div>
    );
  }

  render() {
    return (
      <div className="results-div" hidden={!this.state.isShow}>
        {!this.state.victory ? this.renderWin() : this.renderLose()}
      </div>
    );
  }
}

export default WinLosePage;
