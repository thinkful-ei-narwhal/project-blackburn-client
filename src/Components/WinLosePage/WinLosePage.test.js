import React from 'react';
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import WinLosePage from './WinLosePage';
import BlackBurnContext from '../../Context/BlackburnContext';

describe(`Renders win lose page to dom without crashing`, () => {
    it('renders checkpoint win crashing', () => {
        const wrapper = shallow(
        <BlackBurnContext.Provider>
            <WinLosePage condition = {'checkpoint'} />
        </BlackBurnContext.Provider>
    )
    .find('.results-victory')
    expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders level win crashing', () => {
        const wrapper = shallow(
        <BlackBurnContext.Provider>
            <WinLosePage condition = {'level_beaten'} />
        </BlackBurnContext.Provider>
    )
    .find('.results victory')
    expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders level lose crashing', () => {
        const wrapper = shallow(
        <BlackBurnContext.Provider>
            <WinLosePage condition = {'lose'} />
        </BlackBurnContext.Provider>
    )
    .find(".results defeat")
    expect(toJson(wrapper)).toMatchSnapshot()
    })
})