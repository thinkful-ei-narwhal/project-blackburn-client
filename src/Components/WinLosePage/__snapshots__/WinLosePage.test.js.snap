// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders win lose page to dom without crashing renders checkpoint win crashing 1`] = `null`;

exports[`Renders win lose page to dom without crashing renders level lose crashing 1`] = `null`;

exports[`Renders win lose page to dom without crashing renders level win crashing 1`] = `null`;

exports[`Renders win lose page to dom without crashing renders without crashing 1`] = `null`;
